packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -113,75 +111,39 @@ export async function generateArmTemplate(ctx: SolutionContext): Promise<Result<
   return ok(undefined); // Nothing to return when success
 }
 
-export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<void, FxError>> {
-  ctx.logProvider?.info(
-    format(getStrings().solution.StartDeployArmTemplateNotice, PluginNames.SOLUTION)
+export async function doDeployArmTemplates(ctx: SolutionContext): Promise<Result<void, FxError>> {
+  const progressHandler = await ProgressHelper.startDeployArmTemplatesProgressHandler(
+    getPluginContext(ctx, PluginNames.SOLUTION)
   );
-  const pluginCtx = getPluginContext(ctx, PluginNames.SOLUTION);
-  const progressHandler = await ProgressHelper.startDeployArmTemplatesProgressHandler(pluginCtx);
-  await progressHandler?.next(DeployArmTemplatesSteps.DeployArmTemplates);
+  await progressHandler?.next(DeployArmTemplatesSteps.ExecuteDeployment);
 
-  const azureInfraDir = path.join(ctx.root, baseFolder);
   generateResourceName(ctx);
 
   // update parameters
-  const parameterDefaultFilePath = path.join(
-    azureInfraDir,
-    parameterFolder,
-    parameterDefaultFileName
-  );
-  const parameterTemplateFilePath = path.join(
-    azureInfraDir,
-    parameterFolder,
-    parameterTemplateFileName
-  );
-  let parameterTemplate, parameterJsonString;
-  try {
-    await fs.stat(parameterDefaultFilePath);
-    parameterTemplate = await fs.readFile(parameterDefaultFilePath, ConstantString.UTF8Encoding);
-    parameterJsonString = expandParameterPlaceholders(ctx, parameterTemplate);
-  } catch (err) {
-    try {
-      parameterTemplate = await fs.readFile(parameterTemplateFilePath, ConstantString.UTF8Encoding);
-      parameterJsonString = expandParameterPlaceholders(ctx, parameterTemplate);
-      await fs.writeFile(parameterDefaultFilePath, parameterJsonString);
-    } catch (err) {
-      return err(
-        returnSystemError(
-          new Error(`${parameterTemplateFilePath} does not exist.`),
-          PluginNames.SOLUTION,
-          SolutionError.FailedToDeployArmTemplatesToAzure
-        )
-      );
-    }
-  }
-
-  const parameterJson = JSON.parse(parameterJsonString);
+  const parameterJson = await getParameterJson(ctx);
 
   const resourceGroupName = ctx.config.get(GLOBAL_CONFIG)?.getString(RESOURCE_GROUP_NAME);
   if (!resourceGroupName) {
-    throw returnSystemError(
-      new Error("Failed to get resource group from project solution settings."),
-      PluginNames.SOLUTION,
-      SolutionError.NoResourceGroupFound
-    );
+    throw new Error("Failed to get resource group from project solution settings.");
```

Comment:
Why not throw FxError?

Reply:
Since all error thrown from `doDeployArmDeployment()` method is of the same type `SolutionError.FailedToDeployArmTemplatesToAzure` (`FxError`), in `DeployArmDeployment()` we will wrap all native Error thrown from `doDeployArmDeployment()`.